### PR TITLE
Add BuyNewFlower endpoint

### PR DESCRIPTION
**Pull request by yukari-n-erb**
## Description

Add BuyNewFlower endpoint for implementing store feature.

- API schema
https://www.notion.so/Planning-Product-Development-01f9eff662f24a4fa9adefc6c4e70998#9feb59c163d54d3ca8b5f6f7f4828f33

## Closes issue(s)

If there any issue that is fixed with this PR, please attach it/them here...

## Screenshots

Screen Shot 2021-09-06 at 20 03 19
Screen Shot 2021-09-06 at 20 03 44
Screen Shot 2021-09-06 at 20 03 57

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [x] I have tested this code

### CODE REVIEW

**Comment by yukari-n-erb** [comment]
memo : 

How can I update flower's array ... 🤔

Screen Shot 2021-09-06 at 13 09 11

File: controllers/v1/flowersStore/flowersStore_controller.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	defer ctxCancel()
+
+	collection := mongo_connection.MongoCollection(client, "users")
+
+	// TODO : should error check
+	userData, _ := users_controllers.GetUserByUserId(flowersStore.FireBaseUserId)
+	updatedUser := bson.M{
+		"$set": bson.M{
+			"numCoins":          userData.NumCoins - flowersStore.Price,
+			"flowerCollections": append(userData.FlowerCollections, flowersStore.FlowerID),
```
Comment (yukari-n-erb) [inline controllers/v1/flowersStore/flowersStore_controller.go:28]: memo : 

How can I update flower's array ... 🤔

Screen Shot 2021-09-06 at 13 09 11

File: controllers/v1/flowersStore/flowersStore_controller.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	defer ctxCancel()
+
+	collection := mongo_connection.MongoCollection(client, "users")
+
+	userData, _ := users_controllers.GetUserByUserId(flowersStore.FireBaseUserId)
+
+	updatedUser := bson.M{
+		"$set": bson.M{
+			"numCoins":          userData.NumCoins - flowersStore.Price,
+			"flowerCollections": append(userData.FlowerCollections, flowersStore.FlowerID),
```
Comment (kome12) [inline controllers/v1/flowersStore/flowersStore_controller.go:26]: Can you add the lastUpdate time as well?

File: routes/v1/flowersStore/flowersStore_router.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	utils.EnableCors(&w)
+
+	var flowersStore flowersStore_models.FlowersStore
+	_ = json.NewDecoder(r.Body).Decode(&flowersStore)
+
+	_, err := flowersStore_controllers.BuyNewFlower(flowersStore)
+	if err != nil {
+		var iceCreamError errors_models.IceCreamErrors
+		iceCreamError.Error = err.Error()
+		iceCreamError.Info = "Invalid gardenId provided"
```
Comment (kome12) [inline routes/v1/flowersStore/flowersStore_router.go:25]: Can you update the error reason? I'm guessing it'll be something like "Could not buy flower" or something like that.

**Review by kome12 — COMMENTED** [review]
@yukari-n-erb Overall, I think it looks good. Can you update the code based on the feedback I made?
And I'm guessing you're going to working on the coin check in a separate PR? If so, I created the issue #94 as a reminder.

File: controllers/v1/flowersStore/flowersStore_controller.go (hunk 2 above)
Comment (yukari-n-erb) [reply controllers/v1/flowersStore/flowersStore_controller.go]: I fetched develop data and added lastUpdate logic.

File: routes/v1/flowersStore/flowersStore_router.go (hunk 3 above)
Comment (yukari-n-erb) [reply routes/v1/flowersStore/flowersStore_router.go]: Thank you for your check! I changed string to "Invalid flowersStore request provided" 🙏

**Comment by yukari-n-erb** [comment]
@kome12 
Thank you for your quick review!

I want to confirm coin balance logic. I though that frontend will check this before call `flowersStore` endpoint at the Saturday discussion.  (but It may my misunderstood...?😅 I'd like to confirm this logic again 🙇‍♀️)

**Review by kome12 — APPROVED** [review]
LGTM!

**Comment by kome12** [comment]
> @kome12
> Thank you for your quick review!
> 
> I want to confirm coin balance logic. I though that frontend will check this before call `flowersStore` endpoint at the Saturday discussion. (but It may my misunderstood...?😅 I'd like to confirm this logic again 🙇‍♀️)

The frontend will definitely check this but I think it will be good for the backend to also check this. It doesn't have to be this PR so please go ahead and merge this!

**Comment by yukari-n-erb** [comment]
That's good point, thank you for the advice! I'd like to check #94 🙏

**Pull request merged**